roles/hive/defaults/main.yaml

```diff
@@ -0,0 +1,165 @@
+# Used to cleanup the hive installation
+hive_cleanup: false
+
+hive_repo_address: "https://github.com/ethereum/hive.git"
+hive_repo_version: "master"
+hive_repo_remote: "origin"
+
+hive_systemd_service_name: "hive.service"
+
+hive_dir: "/data/hive"
+hive_dir_bin: "{{ hive_dir }}/bin"
+hive_dir_source: "{{ hive_dir }}/src"
+hive_dir_scripts: "{{ hive_dir }}/scripts"
+hive_dir_results: "{{ hive_dir }}/results"
+hive_dir_web: "{{ hive_dir }}/web"
+
+hive_container_www_name: "nginx-hive"
+hive_container_www_image: "nginx:1.27.3-alpine"
+hive_container_www_env: {}
+hive_container_www_ports:
+  - "127.0.0.1:8080:80"
+hive_container_www_volumes:
+  - "{{ hive_dir_web }}:/usr/share/nginx/html"
+  - "{{ hive_dir_results }}:/usr/share/nginx/html/results"
+hive_container_www_networks: []
+hive_container_www_restart_policy: "always"
+hive_container_www_stop_timeout: 10
+
+
+hive_simulations:
+  # Always pull / build the latest container images
+  always_pull_images: true
+  # Global flags provided to all test runs
+  global_flags:
+    - --docker.nocache=.
+    - --sim.loglevel=3
+    - --loglevel=3
+  # Exclude these containers from cleanup /
+  container_cleanup_exclusions:
+    - telegraf
+    - nginx
+    - logsprout
+    - papertrail
+    - rsyslog
+    - vector
+    - node_exporter
+    - prometheus
+    - grafana
+  # Prune test results older than this value
+  prune_test_results_older_than: 720h # 30 days
+  # Tests to run
+  tests:
+    # Sync tests
+    - simulator: ethereum/sync
+      clients:
+        - besu
+        - erigon
+        - ethereumjs
```

Comment:
where do i configure the client branches and if i want to use a docker image or build from source?

Reply:
You can specify different branches/tags with the client name. For example, if you want branch `abc` from geth, you would write `go-ethereum_abc` .

But I'll add a bit more granual support. Hive has multiple dockerfiles for each client. I'll add support to choose which dockerfile one wants to use. And then depending on the dockerfile, you can also provide build args. 

For example, the default geth [Dockerfile](https://github.com/ethereum/hive/blob/master/clients/go-ethereum/Dockerfile) expects you to have an existing dockerimage with geth inside. It then grabs the binary from there:
```
ARG baseimage=ethereum/client-go
ARG tag=latest
```

But then there's also another [Dockefile.git](https://github.com/ethereum/hive/blob/master/clients/go-ethereum/Dockerfile.git) which will clone a git repo and build geth. This one has the following build args:

```
ARG github=ethereum/go-ethereum
ARG tag=master
```

I was curious to see how we did run the pectra-devnet-4 hive tests.... since I couldn't see specific branches there.  But I figure out that we were running a forked version that contains a client config file in there: https://github.com/danceratopz/hive/blob/prague-devnet-4/configs/prague.yaml

So I'll make it configurable to generate this client file from ansible and provide it to the simulations. With that file we're able to say what dockerfile it should use, and provide it with the build args we want.

Reply:
That would be amazing! thank you :D

Reply:
It's now possible to configure the build args via the client file.